.github/workflows/build.yml

```diff
@@ -4,26 +4,35 @@ on:
     types: [opened, synchronize, edited]
 
 jobs:
+  # We are building "previewctl" outside of "build" because we want to start
+  # the infrastructure job as early as possible (before a full build is done)
+  # and that job requires previewctl
   previewctl:
     if: ${{ contains(github.event.pull_request.body, '[x] /werft with-github-actions') && contains(github.event.pull_request.body, '[x] /werft with-preview') }}
     runs-on: [self-hosted]
     container:
       image: eu.gcr.io/gitpod-core-dev/dev/dev-environment:mads-leeway-v0.7.3.4
     steps:
       - uses: actions/checkout@v3
-      - name: Configure workspace
-        run: cp -r /__w/gitpod/gitpod /workspace
       - name: Build previewctl
         shell: bash
-        working-directory: /workspace/gitpod
+        env:
+          HOME: /home/gitpod
+          PREVIEW_ENV_DEV_SA_KEY: ${{ secrets.GCP_CREDENTIALS }}
         run: |
-          leeway run dev/preview/previewctl:install --dont-test
-          cp /workspace/bin/previewctl /__w/gitpod/gitpod/previewctl
-      - name: "Upload Installer artifacts"
-        uses: actions/upload-artifact@v3
-        with:
-          name: previewctl
-          path: previewctl
+          # Authenticate with GCP so we can use the Leeway cache
+          export PREVIEW_ENV_DEV_SA_KEY_PATH="$HOME/.config/gcloud/preview-environment-dev-sa.json"
+          echo "${PREVIEW_ENV_DEV_SA_KEY}" > "${PREVIEW_ENV_DEV_SA_KEY_PATH}"
+          gcloud auth activate-service-account --key-file "${PREVIEW_ENV_DEV_SA_KEY_PATH}"
+
+          export LEEWAY_WORKSPACE_ROOT="$(pwd)"
+          leeway build dev/preview/previewctl:cli
```

Comment:
Before this change it would build previewctl on every run because it wasn't singed in to GPC so it couldn't access the cache ☺️ I changed the invocation from `leeway run dev/preview/previewctl:install --dont-test` to `leeway build dev/preview/previewctl:cli` because `previewctl:install` is currently told to use `--cache=remote-pull` which means it won't push the cache, so the infra step would have to build previewctl again.

But all of this is changing in a little bit anyway as we'll consume previewctl from the docker image so this entire job is going away (will be replaced by something else) ☺️